public/components/metrics/sidebar/sidebar.tsx

```diff
@@ -5,43 +5,124 @@
+  setSelectedOTIndex,
+}: SideBarMenuProps) => {
   const dispatch = useDispatch();
-
-  const availableMetrics = useSelector(availableMetricsSelector);
+  const [availableOTDocuments, setAvailableOTDocuments] = useState([]);
+  const availableOTDocumentsRef = useRef();
+  availableOTDocumentsRef.current = availableOTDocuments;
+  const promethuesMetrics = useSelector(availableMetricsSelector);
   const selectedMetrics = useSelector(selectedMetricsSelector);
+  const otelIndices = useSelector(otelIndexSelector);
 
   useEffect(() => {
     batch(() => {
       dispatch(loadMetrics());
     });
-  }, [dispatch]);
+  }, [dispatch, selectedDataSource]);
+
+  useEffect(() => {
+    batch(() => {
+      dispatch(loadOTIndices());
+    });
+  }, [dispatch, selectedDataSource]);
+
+  useEffect(() => {
+    if (selectedOTIndex.length > 0 && selectedDataSource[0]?.label === 'OpenTelemetry') {
+      const fetchOtelDocuments = async () => {
+        try {
+          const documentNames = await fetchOpenTelemetryDocumentNames(selectedOTIndex[0]?.label)();
+          const availableOtelDocuments = documentNames?.aggregations?.distinct_names?.buckets.map(
+            (item: any) => {
+              return {
+                id: item.key,
+                name: item.key,
+                catalog: 'OpenTelemetry',
+                type: 'Histogram',
+                index: selectedOTIndex[0]?.label,
+              };
+            }
+          );
+          setAvailableOTDocuments(availableOtelDocuments);
+          dispatch(setMetrics(availableOtelDocuments));
+          dispatch(setDataSourceIcons(coloredIconsFrom(['OpenTelemetry'])));
+        } catch (error) {
+          console.error('Error fetching OpenTelemetry documents:', error);
+        }
+      };
 
-  const handleAddMetric = (metric: any) => dispatch(selectMetric(metric));
+      fetchOtelDocuments();
+    }
+  }, [dispatch, selectedDataSource, selectedOTIndex]);
+
+  const indexPicker = useMemo(() => {
+    const isOpenTelemetry = selectedDataSource[0]?.label === 'OpenTelemetry' ? true : false;
+    if (isOpenTelemetry) {
+      return <IndexPicker otelIndices={otelIndices} setSelectedOTIndex={setSelectedOTIndex} />;
+    }
+  }, [selectedDataSource]);
+
+  const availableMetrics = useMemo(() => {
+    if (selectedDataSource[0]?.label === 'OpenTelemetry' && selectedOTIndex.length > 0)
+      return promethuesMetrics;
+    else if (selectedDataSource[0]?.label === 'Prometheus') return promethuesMetrics;
+    else return [];
+  }, [promethuesMetrics, selectedDataSource, availableOTDocuments, selectedOTIndex]);
+
+  const handleAddMetric = (metric: any) => {
+    console.log('handle add metrics: ', metric);
+    dispatch(selectMetric(metric));
+  };
 
   const handleRemoveMetric = (metric: any) => {
+    console.log('handle remove metrics: ', metric);
     dispatch(deSelectMetric(metric));
   };
 
   return (
     <I18nProvider>
       <section className="sidebarHeight">
+        <DataSourcePicker
```

Comment:
You'll need to look at new Sidebar.  Largely, this component, with no attributes, is what will go here.  None of the change above will be necessary -- all the functionality is in metricSlice!